core-plugins/src/main/java/co/cask/hydrator/plugin/batch/sink/SnapshotFileBatchSink.java

```diff
@@ -76,6 +77,12 @@ public void onRunFinish(boolean succeeded, BatchSinkContext context) {
     super.onRunFinish(succeeded, context);
     if (succeeded) {
       try {
+        if (config.getCleanPartitionsOlderThan() != null) {
```

Comment:
I think this should happen after the onSuccess() call. If an exception occurs when cleaning up old partitions, it would leave things in a weird state.